clients/drcachesim/tests/scheduler_unit_tests.cpp

```diff
@@ -3911,6 +3911,136 @@ test_random_schedule()
+        if (status == record_scheduler_t::STATUS_OK) {
+            if (record.type != expect_type) {
+                std::cerr << "Expected type " << expect_type << " != " << record.type
+                          << "\n";
+                assert(false);
+            }
+            if (expect_addr != 0 && record.addr != expect_addr) {
+                std::cerr << "Expected addr " << expect_addr << " != " << record.addr
+                          << "\n";
+                assert(false);
+            }
+        }
+    };
+    // Advance cpu0 on TID_A to its 1st context switch.
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_THREAD, TID_A);
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_PID, PID_A);
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_MARKER);
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_MARKER);
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_ENCODING);
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_INSTR);
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_MARKER);
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_MARKER);
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_MARKER);
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_MARKER);
+    // Ensure the context switch is *before* the encoding.
+    // Advance cpu0 on TID_B to its 1st context switch.
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_THREAD, TID_B);
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_PID, PID_B);
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_MARKER);
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_MARKER);
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_ENCODING);
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_INSTR);
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_ENCODING);
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_INSTR);
+    // Ensure the switch is *before* the encoding.
+    // TODO i#6635: Insert a tid,pid pair as otherwise the switch is invisible
+    // at the trace_entry_t record level.
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_ENCODING); // A
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_INSTR);    // A
+    // TODO i#6635: Insert a tid,pid pair as otherwise the switch is invisible
+    // at the trace_entry_t record level.
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_ENCODING);    // B
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_INSTR);       // B
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_INSTR);       // B
+    check_next(stream0, record_scheduler_t::STATUS_OK, TRACE_TYPE_THREAD_EXIT); // B
+    // TODO i#6635: Insert a tid,pid pair as otherwise the switch is invisible
```

Comment:
No, it is actually not possible to detect in the resulting core-sharded trace without the tid,pid.

Reply:
Oh right. Because these core-shards are being read from disk (as opposed to the scheduler generating them dynamically from regular traces), so the markers would be the only way to indicate context switches.